cli/cmd/lib_cluster_config.go

```diff
@@ -147,6 +147,45 @@ func getInstallClusterConfig(awsClient *aws.Client, clusterConfigFile string, di
 	return clusterConfig, nil
 }
 
+func getConfigureClusterConfig(awsClient *aws.Client, stacks clusterstate.ClusterStacks, cachedClusterConfig clusterconfig.Config, newClusterConfigFile string, disallowPrompt bool) (*clusterconfig.Config, clusterconfig.ConfigureChanges, error) {
+	newUserClusterConfig := &clusterconfig.Config{}
+
+	err := readUserClusterConfigFile(newUserClusterConfig, newClusterConfigFile)
+	if err != nil {
+		return nil, clusterconfig.ConfigureChanges{}, err
+	}
+
+	promptIfNotAdmin(awsClient, disallowPrompt)
+
+	newUserClusterConfig.Telemetry = isTelemetryEnabled()
+	cachedClusterConfig.Telemetry = newUserClusterConfig.Telemetry
+
+	configureChanges, err := newUserClusterConfig.ValidateOnConfigure(awsClient, cachedClusterConfig)
+	if err != nil {
+		err = errors.Append(err, fmt.Sprintf("\n\ncluster configuration schema can be found at https://docs.cortex.dev/v/%s/", consts.CortexVersionMinor))
+		return nil, clusterconfig.ConfigureChanges{}, errors.Wrap(err, newClusterConfigFile)
+	}
+
+	// intersect with the stale eks node groups
+	eksNodeGroupsToRemove := []string{}
+	staleEKSNgs, staleEKSNgAvailabilities := stacks.GetStaleNodeGroupNames(*newUserClusterConfig)
+	for i := range staleEKSNgs {
+		if slices.HasString(configureChanges.NodeGroupsToRemove, staleEKSNgs[i]) {
+			eksNodeGroupsToRemove = append(eksNodeGroupsToRemove, clusterstate.GetStackName(newUserClusterConfig.ClusterName, staleEKSNgAvailabilities[i], staleEKSNgs[i]))
+		}
+	}
+	configureChanges.NodeGroupsToRemove = eksNodeGroupsToRemove
```

Comment:
Don't reassign NodeGroupsToRemove. Consider adding a fourth field to configureChanges which is dedicated to removing stacks that are not mentioned in both the new and old config. Also consider updating the NodeGroups* fields to clusterconfig.Nodegroup type.